framework/retryresource/resource.go

```diff
@@ -85,7 +90,7 @@ func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
 	}
 
 	n := func(err error, dur time.Duration) {
-		r.logger.Log("warning", fmt.Sprintf("retrying 'GetCurrentState' due to error (%s)", err.Error()))
+		r.logger.Log("warning", fmt.Sprintf("retrying 'GetCurrentState' of resource '%s' due to error (%s)", r.Origin().Name(), err.Error()))
```

Comment:
Random brain wave: maybe we should set `r.logger = r.logger.With("name", r.Origin().Name())` somewhere in `New`.